src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,64 @@
+package duke;
+
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.exception.InsufficientAruments;
+import duke.exception.UnknownCommand;
+import duke.task.TaskList;
+
+/**
+ * Implements the parsing of command line arguments, and create and put the corresponding command with the parsed input
+ * arguments into the command queue.
+ */
+public class Parser {
+    private static CommandQueue queue = new CommandQueue();
+
+    /**
+     * Creates a command according to the user input, parse it,
+     * and pass the relevant input into the command created,
+     * and store the command in the command queue.
+     * @param input The CLI input from the user.
+     * @throws DukeException
+     */
+    public static void createCommand (String input) throws DukeException {
+        switch (input){
+            default : {
+                if (input.matches("mark+ [0-9]+")) {
```

Comment:
Perhaps these if-else statements can be simplified or abstracted out to simplify the code?